client/state/reader/posts/actions.js

```diff
@@ -17,26 +21,92 @@ import analytics from 'lib/analytics';
 import { runFastRules, runSlowRules } from './normalization-rules';
 import Dispatcher from 'dispatcher';
 import { action } from 'lib/feed-post-store/constants';
+import wpcom from 'lib/wp';
 
 function trackRailcarRender( post ) {
 	analytics.tracks.recordEvent( 'calypso_traintracks_render', post.railcar );
 }
 
+function postToKey( post ) {
+	return {
+		feedId: post.feed_ID,
+		postId: post.ID,
+		blogId: ! post.is_external && post.site_ID ? post.site_ID : undefined
+	};
+}
+
+function fetchForKey( postKey ) {
+	let req;
+	if ( postKey.blogId ) {
+		req = wpcom.undocumented().readSitePost( {
+			site: postKey.blogId,
+			postId: postKey.postId
+		} );
+	} else {
+		req = wpcom.undocumented().readFeedPost( postKey );
+	}
+	return req;
+}
```

Comment:
Maybe we could return early here since we don't actually do anything with `req`.

```js
function fetchForKey( postKey ) {
    if ( postKey.blogId ) {
        return wpcom.undocumented().readSitePost( {
            site: postKey.blogId,
            postId: postKey.postId
        } );
    }

    return wpcom.undocumented().readFeedPost( postKey );
}
```